src/DataLib/fendl32B_retrofit/reaction_data.py

```diff
@@ -14,6 +14,67 @@
              'γ'     : array([ 0      ,      0      ])  # gamma emission
 }
 
+def count_emitted_particles(particle, emitted_particle_string):
+    """
+    Count emitted particles from a reaction given a target particle
+        and the particles produced in a neutron activation reaction.
+
+    Arguments:
+        particle (str): Name of the target particle produced in the reaction.
+            Options include n, p, alpha, d, t, and 3He, corresponding to
+            neutrons, protons, alpha particles, deuterons, tritons,
+            and helium-3 nuclides.
+        emitted_particle_string (str): Particle product(s) of the neutron
+            activation, of the format 'p' for a single proton for example or
+            '2n' for two neutrons etc.
+
+    Returns:
+        particle_count (int or None): Count of the target particle present in
+            the product. If particles not present, returns None rather than 0.
+    """
+
+    particle_index = emitted_particle_string.find(particle)
+    number_str = ''
+    for i in range(particle_index - 1, -1, -1):
+        if emitted_particle_string[i].isdigit():
+            number_str = emitted_particle_string[i] + number_str
+        else:
+            break
+
+    if number_str:
+        particle_count = int(number_str)
+    elif particle in emitted_particle_string:
+        particle_count = 1
+    else:
+        particle_count = None
```

Comment:
It might make sense to have this return 0 for this case.  Then you can still use the math without any concern.